### PR TITLE
docs(Scheduler): add docs for async, asap, queue and animationFrame

### PR DESCRIPTION
**Pull request by Podlas29**
**Description:**

Schedulers are probably the most mysterious for newcomers (they were for me and in a way still are).

I believe it would help to show some examples of using schedulers without operators, so that users can
have good intuition what given scheduler can achieve.

I have put descriptions next to already created Schedulers, since these are the ones that should be used primarily.

**Problems:**

* I can not make `{@link async}` in docs, since there is already `async` property on `AjaxRequest` interface. 
* Schedulers and `root` are in the same html file, because they are all variables. Is there a known way to separate them?

**Related issues:**
https://github.com/ReactiveX/rxjs/issues/2379 - question about animationFrame scheduler

### CODE REVIEW

**Comment by staltz** [comment]
@Podlas29 Thank you.

To make it easier to predict how this will look like in the final generated website, could you please host the generated site from your local machine, perhaps to some easy service like http://surge.sh/ or https://zeit.co/now/ ?

**Comment by Podlas29** [comment]
@staltz http://rxjs-docs-prototype.surge.sh/ here you go!

File: src/scheduler/animationFrame.ts [hunk 1]
```diff
@@ -1,4 +1,32 @@
...
+ * Without delay, `animationFrame` scheduler can be used to create smooth browser animations.
+ * It makes sure scheduled task will happen just before next browser content repaint,
+ * thus performing animations as efficiently as possible.
+ *
+ * @example <caption>Schedule div height animation</caption>
+ * const div = document.querySelector('#some-div');
+ *
+ * Rx.Scheduler.schedule(function(height) {
+ *   div.style.height = height + "px"
+ *
```
Comment (blesh) [inline src/scheduler/animationFrame.ts]: Perhaps add a comment here explaining that `this` is a reference to the scheduled `Action`?

**Comment by blesh** [comment]
Looks great, as always, @Podlas29! I've only made one request, but other than that, it looks really good!

**Comment by Podlas29** [comment]
@blesh thanks. I have made changes you requested.

**Pull request merged**